pkg/common/unittestutils/unittestutils.go

```diff
@@ -0,0 +1,101 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package unittestutils
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"sync"
+
+	"sigs.k8s.io/vsphere-csi-driver/pkg/apis/migration"
+	cnsvolume "sigs.k8s.io/vsphere-csi-driver/pkg/common/cns-lib/volume"
+	cnsconfig "sigs.k8s.io/vsphere-csi-driver/pkg/common/config"
+	"sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/common"
+	"sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/common/commonco"
+)
+
+// FakeK8SOrchestrator is used to mock common K8S Orchestrator instance to store FSS values
+type FakeK8SOrchestrator struct {
+	featureStates map[string]string
+}
+
+// GetFakeContainerOrchestratorInterface returns a dummy CO interface based on the CO type
+func GetFakeContainerOrchestratorInterface(orchestratorType int) (commonco.COCommonInterface, error) {
+	if orchestratorType == common.Kubernetes {
+		fakeCO := &FakeK8SOrchestrator{
+			featureStates: map[string]string{
+				"volume-extend": "true",
+				"volume-health": "true",
+				"csi-migration": "true",
+			},
+		}
+		return fakeCO, nil
+	}
+	return nil, fmt.Errorf("Unrecognized CO type")
+
+}
+
+// IsFSSEnabled returns the FSS values for a given feature
+func (c *FakeK8SOrchestrator) IsFSSEnabled(ctx context.Context, featureName string) bool {
+	var featureState bool
+	var err error
+	if flag, ok := c.featureStates[featureName]; ok {
+		featureState, err = strconv.ParseBool(flag)
+		if err != nil {
+			return false
+		}
+		return featureState
+	}
+	return false
+}
+
+// volumeMigration holds mocked migrated volume information
+type volumeMigration struct {
+	// volumePath to volumeId map
+	volumePathToVolumeID sync.Map
+	// volumeManager helps perform Volume Operations
+	volumeManager *cnsvolume.Manager
+	// cnsConfig helps retrieve vSphere CSI configuration for RegisterVolume Operation
+	cnsConfig *cnsconfig.Config
+}
+
+// GetFakeVolumeMigrationService returns the mocked VolumeMigrationService
+func GetFakeVolumeMigrationService(ctx context.Context, volumeManager *cnsvolume.Manager, cnsConfig *cnsconfig.Config) (migration.VolumeMigrationService, error) {
+	// fakeVolumeMigrationInstance is a mocked instance of volumeMigration
+	fakeVolumeMigrationInstance := &volumeMigration{
+		volumePathToVolumeID: sync.Map{},
+		volumeManager:        volumeManager,
+		cnsConfig:            cnsConfig,
+	}
+	return fakeVolumeMigrationInstance, nil
+}
+
+// GetVolumeID mocks the method with returns Volume Id for a given Volume Path
+func (volumeMigration *volumeMigration) GetVolumeID(ctx context.Context, volumePath string) (string, error) {
+	return "", nil
```

Comment:
We need to keep mock mapping of volumePath to VolumeID and GetVolumeID, GetVolumePath should return mock mapping instead of empty string.
DeleteVolumeInfo should help removing mock static mapping of volumepath to volumeid.